src/app/components/left-menu/left-menu.component.ts

```diff
@@ -108,17 +105,10 @@ export class LeftMenuComponent implements OnInit {
     this.menuEventEmitter.next(Object.assign({}, this.toggleStates));
   }
 
-  public expandMenu() {
-    this.isLabelDisplayed = !this.isLabelDisplayed;
-    setTimeout(() => {
-      window.dispatchEvent(new Event('resize'));
-    }, 100);
-  }
-
   /** When opening the dialog of layers to share, we specify the visibility status of all
    * layers so that we choose only the displayed ones */
   public displayShare() {
-    this.shareComponent.openDialog(this.layersVisibilityStatus);
+    this.shareComponent.openDialog(this.mapService.mapComponent.visibilityStatus);
```

Comment:
Is there a possibility that  `this.mapService.mapComponent` is undefined ?

Reply:
As soon as the map is loaded, the mapService gets the map component. So in that span of time, it could be undefined. Do you want me to add a check before that line ?

Reply:
no not necessaraly with what you explained